crates/sui-types/src/temporary_store.rs

```diff
@@ -873,7 +879,11 @@ impl<S: GetModule<Error = SuiError, Item = CompiledModule>> GetModule for Tempor
                 obj.data
                     .try_as_package()
                     .expect("Bad object type--expected package")
-                    .deserialize_module(&module_id.name().to_owned())?,
+                    .deserialize_module(
+                        &module_id.name().to_owned(),
+                        ProtocolConfig::get_for_version(self.protocol_version)
```

Comment:
nit: probably better to cache the version once in `TemporaryStore`?  This is a fairly hot path, so it would be better not to run `ProtocolConfig::get_for_version` for every module access -- that function is not so efficient IIRC (it's recursive, linear in the number of versions, and allocates a the `ProtocolConfig` multiple times).